vochain/indexer/archive.go

```diff
@@ -61,17 +61,33 @@ func (idx *Indexer) ImportArchive(archive []*ArchiveProcess) ([]*ArchiveProcess,
 		} else {
 			continue
 		}
-		creationTime := time.Now()
-		if p.StartDate != nil {
-			creationTime = *p.StartDate
+
+		// For backward compatibility, we try to fetch the start/end date from multiple sources.
+		// If not found, we calculate them from the block count and the default block time.
+		startDate := p.ProcessInfo.StartDate
+		if startDate.IsZero() {
+			if p.StartDate != nil {
+				startDate = *p.StartDate
+			} else {
+				// Calculate startDate equal to time.Now() minus defaultBlockTime*p.ProcessInfo.BlockCount
+				startDate = time.Now().Add(-defaultBlockTime * time.Duration(p.ProcessInfo.BlockCount))
+			}
+		}
+		endDate := p.ProcessInfo.EndDate
+		if endDate.IsZero() {
+			// Calculate endDate equal to startDate plus defaultBlockTime*p.ProcessInfo.BlockCount
+			endDate = startDate.Add(defaultBlockTime * time.Duration(p.ProcessInfo.BlockCount))
 		}
+
 		// Create and store process in the indexer database
 		procParams := indexerdb.CreateProcessParams{
 			ID:                nonNullBytes(p.ProcessInfo.ID),
 			EntityID:          nonNullBytes(p.ProcessInfo.EntityID),
-			StartBlock:        int64(height),
-			EndBlock:          int64(height + 1),
-			BlockCount:        int64(1),
+			StartBlock:        int64(p.ProcessInfo.StartBlock),
+			StartDate:         startDate,
+			EndBlock:          int64(p.ProcessInfo.EndBlock),
+			EndDate:           endDate,
```

Comment:
to me it looks somewhat confusing, prone to error (or redundant?), that we're storing both Block and Date... blocks already have a very definite date

Reply:
The whole point for this PR is to be able to store processes from previous (dead) chains. This is the `archive`.
So we don't have a way to estimate the start/endblock on those processes.